### DISCUSSION TITLE
How do i set swagger config using inheritance for controllers

### DISCUSSION THREAD

**Discussion by shivanshtalwar0** [TypedProject/ts-express-decorators#2457] [General]
I am well aware that controllers in Tsed support inheritance but I want to  to make generic base class which would contain basic rest endpoint implementation consuming repository as a constructor argument from child controller but what i am not sure is how would i set generic swagger config for those rest methods with decorators

for example
```ts
class Base {

constructor(protected service:any){}
@Get('/')
@Summary('how do i set this?')
getAll(){
return service.getAll()
}

}
class ChildController extends Base{

@Inject()
service:ChildRepository;
constructor(){
super(this.service)
}

}

```
@Romakita  or maybe if you have some other nifty way to do this i am all ears. Thanks for creating Tsed lots of love!

**Comment by Romakita** [comment]
Just add `@Controller`  decorator on child controller.

see you

**Reply by Romakita** [reply]
You can’t without overriding the method you want to decorate. I suggest you to stay generic on your summary and description in this case.

**Comment by shivanshtalwar** [comment]
No wait maybe i haven't made myself clear, both base and child controller have @controller decorator i am asking how do i set @summary, @returns which is swagger based decorator which can be different for different child controller based on different prisma model and repository

**Reply by Romakita** [reply]
Arf isn’t really possible by design. Decorators need a physical class method. So if you want to customize a description or a summary for a parent method, you have to redeclare the method and call his parent method.
Then you will be able to add decorators. It’s clearly not a good practice.
My suggestion is to stay generic in your description and summary.

see you
